File: CS/E5153/CategoriesChartReport.cs
using System;
using System.Linq;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using DevExpress.XtraReports.UI;

namespace E5153 {
    public partial class CategoriesChartReport : DevExpress.XtraReports.UI.XtraReport {
        NWindEntities db;

        public CategoriesChartReport() {            
            InitializeComponent();
            db = new NWindEntities();
        }

        private void CategoriesChartReport_DataSourceDemanded(object sender, EventArgs e) {
            this.DataSource = from product in db.Categories
                              select product;
        }

        private void xrChartProducts_BeforePrint(object sender, System.Drawing.Printing.PrintEventArgs e) {         
            Category cat = this.GetCurrentRow() as Category;
            xrChartProducts.DataSource = cat.Products;
        }

    }
}

File: VB/E5153/CategoriesChartReport.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Linq
Imports System.Drawing
Imports System.Collections
Imports System.ComponentModel
Imports DevExpress.XtraReports.UI

Namespace E5153
	Partial Public Class CategoriesChartReport
		Inherits DevExpress.XtraReports.UI.XtraReport
		Private db As NWindEntities

		Public Sub New()
			InitializeComponent()
			db = New NWindEntities()
		End Sub

		Private Sub CategoriesChartReport_DataSourceDemanded(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.DataSourceDemanded
			Me.DataSource = _
				From product In db.Categories _
				Select product
		End Sub

		Private Sub xrChartProducts_BeforePrint(ByVal sender As Object, ByVal e As System.Drawing.Printing.PrintEventArgs) Handles xrChartProducts.BeforePrint
			Dim cat As Category = TryCast(Me.GetCurrentRow(), Category)
			xrChartProducts.DataSource = cat.Products
		End Sub

	End Class
End Namespace
